### PR TITLE
FreeBSD binary startup improvements

### PR DESCRIPTION
**Pull request by trasz**
The purpose of those commits is to do one sysctl(2) syscall during binary startup instead of current three.

### CODE REVIEW

**Comment by davidtgoldblatt** [comment]
Looks reasonable to me, but I'll wait for @interwq since he had concerns with the previous diff, too.

Is jemalloc startup time something you/FreeBSD more generally are trying to optimize for, or are you just taking a minor win where you can get one? I bet if we sat down and thought about it we could find some more places.

File: src/pages.c [hunk 1]
```diff
@@ -373,9 +378,19 @@ os_overcommits_sysctl(void) {
 	size_t sz;
 
 	sz = sizeof(vm_overcommit);
+#if defined(__FreeBSD__) && defined(VM_OVERCOMMIT)
+	int mib[2];
+
+	mib[0] = CTL_VM;
+	mib[1] = VM_OVERCOMMIT;
+	if (sysctl(mib, 2, &vm_overcommit, &sz, NULL, 0) == -1) {
```
Comment (interwq) [inline src/pages.c]: nit: use ```!= 0``` to match the other branch

**Comment by davidtgoldblatt** [comment]
I *think* any update to the commit should trigger another run (I added @trasz to the whitelist).

I opened https://github.com/jemalloc/jemalloc-ci/issues/10; I think we should kill the whitelist.

**Comment by trasz** [comment]
Hm, so what's the preferred course of action here?  Should I just amend the commit to change that comparison, and then "push -f"?

As for the startup time - yeah, it kind of turned out to be a miniproject of sorts, to get rid of all the unnecessary syscalls during binary startup.  I've already got rid of unnecessary path lookups; sysctl-s are pretty much done with this PR.

**Comment by davidtgoldblatt** [comment]
Yeah, that should do it (and should also trigger CI).

I'll try and ping you on an issue if we think of a startup time improvement idea.

**Comment by interwq** [comment]
It looks our CI wasn't triggered @davidtgoldblatt -- do we have to close and re-open the pull request?

**Comment by davidtgoldblatt** [comment]
Looks like the commit wasn't amended yet?

**Comment by interwq** [comment]
I think it has been changed. The previous comment is now covered by "show outdated".

**Comment by davidtgoldblatt** [comment]
Ugggh, left my laptop at work and don't have an appropriate key at home. Sorry, @trasz, I'll try to fix things first thing Monday (or test manually over the weekend).

**Comment by trasz** [comment]
Ping?

**Comment by davidtgoldblatt** [comment]
Hi @trasz, apologies again for the amount of dumb stuff you've had to wade through for this. I killed the whitelist entirely for our CI; could you try one more time to tweak the commit somehow and amend?

**Comment by trasz** [comment]
Okay.  Is it ok now?

**Comment by davidtgoldblatt** [comment]
Not sure why CI still isn't triggering, but I tested manually. Merging.

**Pull request merged**